fix: CharacterOverlay.tsx の useRef に @types/react@19 対応の初期値を追加する

## 概要

`src/renderer/src/CharacterOverlay.tsx` の `useRef<T>()` 呼び出しが `@types/react@19` の型定義変更により TypeScript エラーになっている。

## エラー内容

```
src/renderer/src/CharacterOverlay.tsx(18,27): error TS2554: Expected 1 arguments, but got 0.
src/renderer/src/CharacterOverlay.tsx(19,24): error TS2554: Expected 1 arguments, but got 0.
```

## 該当箇所

`src/renderer/src/CharacterOverlay.tsx:18-19`

```typescript
const displayTimerRef = useRef<ReturnType<typeof setTimeout>>();
const fadeTimerRef = useRef<ReturnType<typeof setTimeout>>();
```

## 原因

`@types/react@19` で `useRef<T>()` を引数なしで呼ぶと `RefObject<T>`（読み取り専用）が返るように型定義が変更された。`MutableRefObject<T | undefined>` を得るには `undefined` を明示的に渡す必要がある。

## 修正方法

```typescript
const displayTimerRef = useRef<ReturnType<typeof setTimeout>>(undefined);
const fadeTimerRef = useRef<ReturnType<typeof setTimeout>>(undefined);
```

## 補足

- `npm run build`（electron-vite）はエラーなく通過する（vite は型チェックを行わないため）
- `npm test`（vitest）も通過する
- `tsc --noEmit` を実行すると顕在化する既存バグ
- PR #30（Issue #26 対応）のコードレビュー中に発見

--- a/src/renderer/src/CharacterOverlay.tsx
+++ b/src/renderer/src/CharacterOverlay.tsx
@@ -10,8 +10,8 @@ export const CharacterOverlay: React.FC = () => {
   const [lottieData, setLottieData] = useState<object | null>(null);
   const [characterFile, setCharacterFile] = useState('dance.json');
   const displayDurationRef = useRef(5000);
-  const displayTimerRef = useRef<ReturnType<typeof setTimeout>>();
-  const fadeTimerRef = useRef<ReturnType<typeof setTimeout>>();
+  const displayTimerRef = useRef<ReturnType<typeof setTimeout>>(undefined);
+  const fadeTimerRef = useRef<ReturnType<typeof setTimeout>>(undefined);
 
   const applySettings = useCallback((settings: AppSettings) => {
     setCharacterFile(settings.characterFile);

